### ISSUE TITLE
Update / Create related entities

### ISSUE THREAD

**Issue by ionut-tanasa** [rinvex/repository#51]
Hey, me again :) Just throwing some ideas here...

It would be cool to have some kind of mechanism to allow us to save  / create related models.

Given this payload: 

``` json
{
  "user": {
    "email" : "me@example.com",
    "profile": {
      "name": "John",
      "surname": "Doe"
    }
  }
}
```

Calling `$repository->create($data);` the following things should happen: 
- create the user
- create the profile
- associate the profile to the user

The `profile` key, obviously, would be the name of the relation on the `User` entity.

Same for the `update()`. If `user.id` and `profile.id` have values the repository would updated them accordingly.

Do you think this would be doable?

**Comment by Omranic** [comment]
Yeah, another cool idea. Honestly I've already thought about it before, but still didn't come to the proper way to implement it. Will give it a shot soon. Thank you @ionut-tanasa 👍 Keep the good ideas flowing 😉

**Comment by Omranic** [comment]
This is a priority now 😃

**Comment by Omranic** [comment]
I'm working on this right now, here's initial implementation if you've any comments or suggestions:

Usage (inside both `create` & `update` methods within `EloquentRepository` we'll call the following):
```php
        // Sync relationshipts
        $this->relate($this->extractRelations($attributes, $entity));
```

Core implementation:
```php
    /**
     * Extract relationships.
     *
     * @param array $attributes
     * @param       $entity
     *
     * @return array
     */
    protected function extractRelations(array $attributes, $entity)
    {
        $relations = [];
        $potential = array_diff(array_keys($attributes), $entity->getFillable());

        array_walk($potential, function ($item) use ($entity, &$relations) {
            if (method_exists($entity, $item)) {
                $relations[$item] = get_class($entity->$item());
            }
        });

        return $relations;
    }

    /**
     * Sync relationships.
     *
     * @param array $attributes
     * @param       $entity
     *
     * @return void
     */
    protected function relate(array $relations)
    {
        foreach ($relations as $method => $relation) {
            switch ($relation) {
                case 'Illuminate\Database\Eloquent\Relations\BelongsToMany':
                default:
                    // Do something
            }
        }
    }
```

**Comment by Omranic** [comment]
**Note:** This implementation handles syncing existing relations, but doesn't handle creation of related entities. Honestly this is what we currently need in our internal projects, and we think it's fair enough as for now, so will keep it as simple as that. If anyone would like to add support for creation, please free to submit a PR 😉

**Comment by Omranic** [comment]
Implemented in 880ad5615fd5659cfbf43ab28a82bac5024d7d9a
